### ISSUE TITLE
[bug] Randomly errors “connection to dock not established” on handset

### ISSUE THREAD

**Issue by ascoarchitect** [unfoldedcircle/bug-reports#190]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Handset reports “connection to dock not established” for IR actions after it’s not been used for a few minutes. Handset is within 1m of the dock so well within range. Takes 10s to become responsive again then works fine.

### How to Reproduce

Leave handset for 5 minutes
Try to use any of the buttons
Error reported on handset and IR commands not fired

### Expected behavior

Handset to be responsive and operational at all times.

### System version

V1.4.5

### What part of the system affected by the problem?

Overall system

**Comment by kennymc-c** [comment]
Although 10 seconds is quite long this probably happens because the remote activates a sleep mode when it's not in the dock and has to reconnect to the dock after it wakes up. You can define the sleep timeout in the settings or prevent the remote from going into sleep mode during an activity. For me it doesn't take that long and maybe its why you get an error message but this could related to the WiFi network.

**Comment by leoschweizer** [comment]
Basically any command involving IR or Bluetooth will fail for me right after waking up the remote. Which is quite annoying because the basic use case of watching a film for two hours and then picking up the remote to power off just doesn't work.

If the connection can't be reestablished quickly enough to be immediate, at least the remote should _wait_ for a connection for a reasonable amount of time before just failing (and this doesn't take anywhere near 10 seconds in my case).

Disabling sleep entirely during activities just to make that very basic use case work properly sounds like a bad joke to me.

**Comment by ascoarchitect** [comment]
100% agree with you @leoschweizer - whilst I accept there is a power saving benefit for the remote going to sleep, this should exclude IR functions at the very least as the basic function will be to point at a device and fire a command. I can (slightly) understand it for integrations with HA or Hue for example, but for me this is a fundamental flaw.

If users of the remote are cursing at it and reverting back to using one of the many other IR-based remotes for carrying out a function like volume control or power then this product has failed in its goal to be a replacement. Most (if not all?) of these remote will have a charging dock which will be used most (if not all?) evenings after use.

If the remote cannot last 1 day without needing a charge after staying awake for IR/BT functions then I question whether it's actually useful. Expecting the user to configure this as a 'bypass' seems to be something that requires remediation.

**Comment by kennymc-c** [comment]
I also agree with you. There maybe could be a timeout after the remote wakes up that waits to execute any commands that need a dock or ip connection. After the connection has been established the commands are automatically executed. Maybe it should also ignore commands that were entered multiple times during the timeout.
There is already something similar for activities but it still requires you to restart the activity when all needed connections have been established.

**Comment by tastyrepublic** [comment]
After updating to the latest firmware, I randomly disconnected to the dock too…and need to reboot the dock. It’s just can’t connect to the dock after wake up

**Comment by vistalba** [comment]
I run into this issue almost all days multiple times.
As soon as the remote was in sleep mode. When I then pick it up it just need 5-10 seconds until anything is working. 
It is annoying as this is my usual use case of a remote. May I already watching a movie for 60 minutes and press "pause" for a short break it does not work.

**Comment by jantoney** [comment]
This might help everyone here.

If you're running this from an activity, then when configuring the activity there is a "Prevent sleep" option top left of the configure activity screen. 
The explanation of the option is "While the activity is on, prevent the remote to go to sleep. Note: battery life will be reduced."

You might just need to pop the remote back on the dock a little more often, but it should resolve your issues.
